book/src/user/serialization.md

```diff
@@ -0,0 +1,60 @@
+# Serialization Format
+
+With the `serialization` feature, which is enabled by default, all structs that
+need to communicated will have `serialize()` and `deserialize()` methods.
+
+The format is basically the `serde` encoding of the structs using the
+[`postcard`](https://docs.rs/postcard/latest/postcard/) crate.
+
+- Integers are encoded in [varint
+  format](https://postcard.jamesmunns.com/wire-format#varint-encoded-integers)
+- Fixed-size byte arrays are encoded as-is (e.g. scalars, elements)
+  - Note that the encoding of scalars and elements are defined by the
+    ciphersuites.
+- Variable-size byte arrays are encoded with a length prefix (varint-encoded)
+  and the array as-is (e.g. the message)
+- Maps are encoded as the varint-encoded item count, followed by concatenated
+  item encodings.
+- Ciphersuite IDs are encoded as the 4-byte CRC-32 of the ID string.
```

Comment:
If the IDs are encoded as the 4-byte CRC-32 of a string, does that make it more fine to lean on the spec contextString parameter instead?

Reply:
I simply hadn't though of that, good idea. The only catch is that we'd need to change the `ID` in the Ciphersuite to be the contexString instead of the ID (and maybe change its name to `CONTEXT_STRING` too?) - otherwise we'd need to add yet another field to the ciphersuite which doesn't seem great. What do you think?

Reply:
I'll revisit this in https://github.com/ZcashFoundation/frost/issues/534